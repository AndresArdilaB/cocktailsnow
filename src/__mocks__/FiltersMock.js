const FiltersMock = [
  { label: 'Categories', param: 'category', items: [{ value: 'Test', key: 'test' }] },
  { label: 'Glasses', param: 'glass', items: [{ value: 'Test', key: 'test' }] },
  { label: 'Ingredients', param: 'ingredient', items: [{ value: 'Test', key: 'test' }] },
  { label: 'Type', param: 'type', items: [{ value: 'Test', key: 'test' }] },
];

export default FiltersMock;