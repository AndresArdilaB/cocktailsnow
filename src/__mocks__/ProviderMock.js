import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
