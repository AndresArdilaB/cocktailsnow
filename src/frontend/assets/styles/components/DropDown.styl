.container 
  background-color: rgba(255, 255, 255, 0.1)
  margin: 15px 10px
  border: 2px solid white
  padding: 0px 20px
  border-radius: 25px

.select 
  width: 205px
  background-color: transparent
  color: white
  font-family: "Muli", sans-serif
  font-size: 16px
  height: 50px
  border: none

::placeholder 
  color: white
