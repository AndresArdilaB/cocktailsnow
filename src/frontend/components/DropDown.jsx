import React from 'react';
import '../assets/styles/components/DropDown.styl';

const DropDown = ({ items, label, param, callBack }) => {

  return (
    <div className='container'>
      <select className='select' onChange={(e) => callBack(e, param)}>
        <option value=''>{label}</option>
        {
          items
            .filter((f) => f.value !== '')
            .map((d, i) => (
              <option key={i.toString()} id={d.key} value={d.key}>
                {d.value}
              </option>
            ))
        }
      </select>
    </div>
  );
};

export default DropDown;
