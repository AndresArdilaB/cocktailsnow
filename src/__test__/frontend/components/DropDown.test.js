/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import FiltersMock from '../../../__mocks__/FiltersMock';
import DropDown from '../../../frontend/components/DropDown';

describe('<DropDown />', () => {

  test('Component render', () => {
    const dropDown = shallow(
      <ProviderMock>
        <DropDown />
      </ProviderMock>,
    );

    expect(dropDown.length).toEqual(1);
  });

  test('Test options list', () => {
    const wrapper = mount(
      <ProviderMock>
        <DropDown
          items={FiltersMock[0].items}
          label={FiltersMock[0].label}
          param={FiltersMock[0].param}
          callBack={() => { }}
        />
      </ProviderMock>,
    );

    expect(wrapper.find('[id="test"]')).toHaveLength(1);
  });

});

describe('DropDown Snapshot', () => {
  test('Testing render component UI', () => {
    const dropDown = create(
      <ProviderMock>
        <DropDown
          items={FiltersMock[0].items}
          label={FiltersMock[0].label}
          param={FiltersMock[0].param}
          callBack={() => { }}
        />
      </ProviderMock>,
    );

    expect(dropDown.toJSON()).toMatchSnapshot();
  });
});
