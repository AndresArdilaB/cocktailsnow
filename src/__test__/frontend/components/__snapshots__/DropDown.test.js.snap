// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Snapshot Testing render component UI 1`] = `
<div
  className="container"
>
  <select
    className="select"
    onChange={[Function]}
  >
    <option
      value=""
    >
      Categories
    </option>
    <option
      id="test"
      value="test"
    >
      Test
    </option>
  </select>
</div>
`;
